### PR TITLE
Detect and report races between non-atomic and atomic accesses

### LINKED ISSUE #94: Properly report atomic accesses as atomic accesses
Currently we model atomic accesses as 1-byte reads. And that's how they appear in race reports. This can confuse whoever reads the report, because source code contains no 1-byte reads. This in turn can lead to a conclusion that the tool produces bogus reports.
We need to properly report atomic accesses at atomic accesses.

### PR DESCRIPTION
**Pull request by xairy**
Resolves #94

### CODE REVIEW

**Comment by xairy** [comment]
After this patch KTSAN reports a lot of races between fs/dcache.c, fs/namei.c and lib/lockref.c

File: mm/ktsan/sync_atomic.c [hunk 1]
```diff
@@ -111,21 +105,17 @@ static kt_tab_sync_t *kt_atomic_post_op(kt_thr_t *thr, uptr_t pc, uptr_t addr,
...
 }
 
-#define KT_ATOMIC_OP(op, ad, mo, read, write)				\
+#define KT_ATOMIC_OP(op, ad, mo, size, read, write)			\
 do {									\
 	kt_tab_sync_t *sync = NULL;					\
 									\
 	sync = kt_atomic_pre_op(thr, pc, ad, mo, read, write, sync);	\
 									\
+	kt_access(thr, pc, ad, size, !write, true);			\
```
Comment (dvyukov) [inline mm/ktsan/sync_atomic.c]: You moved the annotation from where it were, it should lead to false positive reports for some code patterns. Please write a test that detects these false positives.

Comment (xairy) [inline mm/ktsan/sync_atomic.c]: What code patterns do you mean?

Comment (dvyukov) [inline mm/ktsan/sync_atomic.c]: What code patterns have you already considered?

Comment (xairy) [inline mm/ktsan/sync_atomic.c]: Atomic/non-atomic reads/writes concurrent with atomic/non-atomic reads/writes.

Comment (dvyukov) [inline mm/ktsan/sync_atomic.c]: Consider also patterns when you read/write/allocate/free the variable before/after the atomic access.

Comment (xairy) [inline mm/ktsan/sync_atomic.c]: Done.

**Comment by xairy** [comment]
PTAL

**Comment by dvyukov** [comment]
LGTM

**Pull request merged**